### PR TITLE
Rodrigodmd/dir template

### PR DESCRIPTION
**Pull request by rodrigodmd**
The idea is to copy a template replacing the destiny path name similar to the template.
Example:
If context is {tag: 'value'}
the action would be:
   /some/path-{ tag }/..
would be copied to
   /some/path-value/..

### CODE REVIEW

**Comment by SBoudrias** [comment]
I'd be okay making this feature change, but let's use the same template syntax as we do for file templates.

**Comment by rodrigodmd** [comment]
Modified dir template syntax to match file template.
Example:
If context is {tag: 'value'}
the template dir structure would be:
   /some/path-<%= tag %>/..
would copied to:
   /some/path-value/..

**Comment by bryancuster** [comment]
👍 I would love this feature.

File: lib/actions/copy.js [hunk 1]
```diff
@@ -63,5 +63,10 @@ exports._copySingle = function (from, to, options) {
     contents = applyProcessingFunc(options.process, file.contents, file.path);
   }
 
+  for (var key in context) {
+    to = to.replace('<%=' + key + '%>', context[key]);
```
Comment (SBoudrias) [inline lib/actions/copy.js]: It'd be better to use the template engine to compile the path string the same way we compile the files as templates. Using replace is too basic and might creates inconsistent behavior between the path and the file templates themselves.

Comment (rodrigodmd) [reply lib/actions/copy.js]: I agree. Modified implementation to:
to = ejs.render(to, context, tplSettings);

Also added if condition to render only if context is not null.

**Comment by SBoudrias** [comment]
Do we really need to add a new method? Let's just do this by default on the copy methods.

**Comment by rodrigodmd** [comment]
I agree. Removed new action. We will just do this by default on the copy method.

**Comment by SBoudrias** [comment]
Thanks!! I'll prepare a new release soon.

**Comment by asmagin** [comment]
Why not just to pass `pathProcess` function to options? It will allow me to use templates, `regex` or just `string.replace` ?

**Comment by asmagin** [comment]
@SBoudrias `copy` doesn't look like a right place for templates functionality. I've moved it into a `copyTpl` and apply it in the same way as the templates for a content.

**Pull request merged**